Fix remaining smart pointer errors in webpushd

fred-wang opened the pull request:
#### 0157ffa26fb13240224fc8b803e94ca04a315da0
<pre>
Fix remaining smart pointer errors in webpushd
<a href="https://bugs.webkit.org/show_bug.cgi?id=299368">https://bugs.webkit.org/show_bug.cgi?id=299368</a>

Reviewed by Chris Dumez.

Note: The remaining errors for _WKMockUserNotificationCenter.mm is a
false positive due to the checker not recognizing _init*
See <a href="https://github.com/llvm/llvm-project/pull/160569">https://github.com/llvm/llvm-project/pull/160569</a>

* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations: Update expectations.
* Source/WebKit/webpushd/MockPushServiceConnection.mm:
(WebPushD::MockPushServiceConnection::subscribe): Use nil to turn off false positive warning.
* Source/WebKit/webpushd/WebPushDaemon.h: Make m_userNotificationCenterClass a RetainPtr.
* Source/WebKit/webpushd/WebPushDaemon.mm:
(WebPushD::WebPushDaemon::handleIncomingPush): Ditto.
(WebPushD::WebPushDaemon::subscribeToPushService): Ditto.
(WebPushD::WebPushDaemon::showNotification): Ditto. Also wrap an arg in a RetainPtr.
(WebPushD::WebPushDaemon::getNotifications): Make m_userNotificationCenterClass a RetainPtr.
(WebPushD::WebPushDaemon::cancelNotification): Ditto.
(WebPushD::WebPushDaemon::getPushPermissionState): Ditto.
(WebPushD::WebPushDaemon::requestPushPermission): Ditto.
(WebPushD::WebPushDaemon::setAppBadge): Ditto.
(WebPushD::WebPushDaemon::getAppBadgeForTesting): Ditto.
* Source/WebKit/webpushd/WebPushDaemonMain.mm: Use a RetainPtr for the result of NSSearchPathForDirectoriesInDomains.
(WebKit::getWebPushDirectoryPathWithMigrationIfNecessary):
* Source/WebKit/webpushd/_WKMockUserNotificationCenter.mm:
(+[_WKMockUserNotificationCenter _internalInitWithBundleIdentifier:]):
  Use a retainPtr for a DISPATCH_QUEUE_SERIAL_WITH_AUTORELEASE_POOL arg, which calls dispatch_queue_attr_make_with_autorelease_frequency.

Canonical link: <a href="https://commits.webkit.org/300566@main">https://commits.webkit.org/300566@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/461d52c40de753abd8ff89f2fbc2bdd31def68a8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123117 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/42831 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33528 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129777 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/75228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43555 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51426 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/93590 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/75228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126068 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34711 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110180 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74223 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33678 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73292 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104423 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28562 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132505 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50067 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38117 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/102088 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50443 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106401 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/101941 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25898 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47311 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25504 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/46840 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/49922 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/55683 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/49390 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/52742 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/51071 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

fred-wang commented on Source/WebKit/webpushd/_WKMockUserNotificationCenter.mm:
```diff
@@ -33,15 +33,15 @@
 
     if (!centers->get()[bundleIdentifier])
-        centers->get()[bundleIdentifier] = adoptNS([[_WKMockUserNotificationCenter alloc] _internalInitWithBundleIdentifier:bundleIdentifier]).autorelease();
+        centers->get()[bundleIdentifier] = [_WKMockUserNotificationCenter _internalCreateWithBundleIdentifier:bundleIdentifier];
```
This is an attempt to fix the following error reports:

https://build.webkit.org/results/Apple-Sequoia-Safer-CPP-Checks/300477@main%20(5301)/scan-build-output/StaticAnalyzer/WebKit/StaticAnalyzerReports/report-_WKMockUserNotificationCenter.mm-centersByBundleIdentifier-4-6d99fc.html#EndPath

https://build.webkit.org/results/Apple-Sequoia-Safer-CPP-Checks/300477@main%20(5301)/scan-build-output/StaticAnalyzer/WebKit/StaticAnalyzerReports/report-_WKMockUserNotificationCenter.mm-centersByBundleIdentifier-4-589528.html#EndPath

It seems some past commits move calls to `adoptNS()` as soon as possible in order to avoid memory leaks (https://commits.webkit.org/234445@main, https://commits.webkit.org/234468@main, https://commits.webkit.org/234818@main, https://commits.webkit.org/233998@main) so this is what I've tried here but that's the only thing that changed so to be honest I'm not exactly sure why that makes the analyzer happy...

rniwa replied:
Oh, I have a LLVM side fix for this: https://github.com/llvm/llvm-project/pull/160569/files
It's because we don't recognize `_init` as an init function at the moment.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/eb9ca719d52ad4d2463754472258a536ac1f6dac. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51192

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/461d52c40de753abd8ff89f2fbc2bdd31def68a8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51192

fred-wang committed:
Fix remaining smart pointer errors in webpushd
https://bugs.webkit.org/show_bug.cgi?id=299368

Reviewed by Chris Dumez.

Note: The remaining errors for _WKMockUserNotificationCenter.mm is a
false positive due to the checker not recognizing _init*
See https://github.com/llvm/llvm-project/pull/160569

* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations: Update expectations.
* Source/WebKit/webpushd/MockPushServiceConnection.mm:
(WebPushD::MockPushServiceConnection::subscribe): Use nil to turn off false positive warning.
* Source/WebKit/webpushd/WebPushDaemon.h: Make m_userNotificationCenterClass a RetainPtr.
* Source/WebKit/webpushd/WebPushDaemon.mm:
(WebPushD::WebPushDaemon::handleIncomingPush): Ditto.
(WebPushD::WebPushDaemon::subscribeToPushService): Ditto.
(WebPushD::WebPushDaemon::showNotification): Ditto. Also wrap an arg in a RetainPtr.
(WebPushD::WebPushDaemon::getNotifications): Make m_userNotificationCenterClass a RetainPtr.
(WebPushD::WebPushDaemon::cancelNotification): Ditto.
(WebPushD::WebPushDaemon::getPushPermissionState): Ditto.
(WebPushD::WebPushDaemon::requestPushPermission): Ditto.
(WebPushD::WebPushDaemon::setAppBadge): Ditto.
(WebPushD::WebPushDaemon::getAppBadgeForTesting): Ditto.
* Source/WebKit/webpushd/WebPushDaemonMain.mm: Use a RetainPtr for the result of NSSearchPathForDirectoriesInDomains.
(WebKit::getWebPushDirectoryPathWithMigrationIfNecessary):
* Source/WebKit/webpushd/_WKMockUserNotificationCenter.mm:
(+[_WKMockUserNotificationCenter _internalInitWithBundleIdentifier:]):
  Use a retainPtr for a DISPATCH_QUEUE_SERIAL_WITH_AUTORELEASE_POOL arg, which calls dispatch_queue_attr_make_with_autorelease_frequency.

Canonical link: https://commits.webkit.org/300566@main

webkit-commit-queue commented:
Committed 300566@main (0157ffa26fb1): <https://commits.webkit.org/300566@main>

Reviewed commits have been landed. Closing PR #51192 and removing active labels.
